### PR TITLE
Backport minimum commission staking parameter from the cosmos-sdk 

### PR DESCRIPTION
**Pull request by lydia-pierce**
This PR is a combination of [reverting](https://github.com/ChihuahuaChain/chihuahua/pull/4) the code we had previously merged into Chihuahua from Juno, and [adding](https://github.com/pomifer/cosmos-sdk/pull/1) some code from the sdk.

We give credit to @marbar3778 and Binary Holdings for being crucial in this PR.

Key changes
- Bumps to the latest stable sdk version that is slightly modified to backport upcoming changes to the staking module to add a minimum commission fee (sdk changelog [here](https://github.com/cosmos/cosmos-sdk/blob/release/v0.44.x/CHANGELOG.md))
- Adds minimum commission fee as a parameter to the staking module
- Stops validators from creating or changing to a commission fee below that of the minimum commission fee
- Forces validator below the new minimum commission fee to change to the minimum commission fee

Testing in place
- Uses unit tests provided by the backported PR
- Local network: provides instructions and scripts to replicate

### CODE REVIEW

File: testutil/scripts/single-validator-node.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+chihuahuad collect-gentxs
+
+# Set proper defaults and change ports
+sed -i 's#"tcp://127.0.0.1:26657"#"tcp://0.0.0.0:26657"#g' ~/.chihuahua/config/config.toml
+sed -i 's/timeout_commit = "5s"/timeout_commit = "1s"/g' ~/.chihuahua/config/config.toml
+sed -i 's/timeout_propose = "3s"/timeout_propose = "1s"/g' ~/.chihuahua/config/config.toml
+sed -i 's/index_all_keys = false/index_all_keys = true/g' ~/.chihuahua/config/config.toml
+
+# Start the chihuahua
+chihuahuad start
```
Comment (evan-forbes) [inline testutil/scripts/single-validator-node.sh:27]: we're happy to remove these scripts, but wanted to provide them for anyone wanting to test a local testnet upgrade

File: app/app.go [hunk 2]
```diff
@@ -650,3 +606,44 @@ func initParamsKeeper(appCodec codec.BinaryCodec, legacyAmino *codec.LegacyAmino
 
 	return paramsKeeper
 }
+
+// RegisterUpgradeHandlers returns upgrade handlers
+func (app *App) RegisterUpgradeHandlers(cfg module.Configurator) {
+	app.UpgradeKeeper.SetUpgradeHandler(v1UpgradeName, func(ctx sdk.Context, plan upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		minCommissionRate := sdk.NewDecWithPrec(5, 2)
+
+		fmt.Println("\nvm:\n", vm)
```
Comment (evan-forbes) [inline app/app.go]: whoops, we should delete this line

File: app/app.go [hunk 3]
```diff
@@ -650,3 +606,44 @@ func initParamsKeeper(appCodec codec.BinaryCodec, legacyAmino *codec.LegacyAmino
 
 	return paramsKeeper
 }
+
+// RegisterUpgradeHandlers returns upgrade handlers
+func (app *App) RegisterUpgradeHandlers(cfg module.Configurator) {
+	app.UpgradeKeeper.SetUpgradeHandler(v1UpgradeName, func(ctx sdk.Context, plan upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		minCommissionRate := sdk.NewDecWithPrec(5, 2)
```
Comment (evan-forbes) [inline app/app.go:612]: we didn't move the 5 out as a constant here, as we would just delete it later. To change in the future, we only have to change the staking parameter.

File: app/app.go (hunk 2 above)
Comment (lydia-pierce) [reply app/app.go]: Done in recent [committ](52b5ef4d0f8a8ce848b556c1e98ad518ca19f8d2)

**Comment by tac0turtle** [comment]
to preface this, this won't bump everyones commission to 5% this still needs to be added to the sdk.

**Comment by evan-forbes** [comment]
> to preface this, this won't bump everyones commission to 5% this still needs to be added to the sdk.

I think we included this in the upgrade handler, it seemed to bump the commission up on our local testnet

edit: and the most recent testnet

**Comment by Highlander-maker** [comment]
Hey, will validators need to upgrade or will the chain continue as is?

**Comment by evan-forbes** [comment]
> Hey, will validators need to upgrade or will the chain continue as is?

Validators will need to upgrade at a specific tbd height, as this upgrade is consensus breaking

**Review by evan-forbes — APPROVED** [review]
testnet was successful!

I think we can merge this PR and create a tagged version after we address any lingering nit picks or questions

**Pull request merged**